jdbc/src/main/java/com/interface21/transaction/support/TransactionSynchronizationManager.java

```diff
@@ -8,16 +10,28 @@ public abstract class TransactionSynchronizationManager {
 
     private static final ThreadLocal<Map<DataSource, Connection>> resources = new ThreadLocal<>();
 
+    static {
+        resources.set(new HashMap<>());
+    }
+
     private TransactionSynchronizationManager() {}
 
     public static Connection getResource(DataSource key) {
-        return null;
+        return resources.get().get(key);
     }
 
     public static void bindResource(DataSource key, Connection value) {
+        if (getResource(key) != null) {
+            throw new DataAccessException("Transaction already started for this datasource");
+        }
+        resources.get().put(key, value);
+    }
+
+    public static void unbindResource(DataSource key) {
+        resources.get().remove(key);
     }
 
-    public static Connection unbindResource(DataSource key) {
-        return null;
+    public static boolean doseNotManage(DataSource dataSource) {
+        return getResource(dataSource) == null;
     }
```

Comment:
```suggestion
    public static boolean doseNotManage(DataSource dataSource) {
        return !resources.get().containsKey(dataSource);
    }
```

이렇게 하는 건 어떨까요?
크게 차이는 없겠지만 무의미하게 다른 메서드를 호출하고 있기도 하고
null로 로직을 처리하는 게 조금 불안하네요 :potato_fear: